pkg/providers/snow/reconciler/clientbuilder.go

```diff
@@ -0,0 +1,97 @@
+package reconciler
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+
+	awstypes "github.com/aws/aws-sdk-go-v2/aws"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	awscredentials "github.com/aws/aws-sdk-go-v2/credentials"
+	"gopkg.in/ini.v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/eks-anywhere/pkg/aws"
+)
+
+type awsClientBuilder struct {
+	client client.Client
+}
+
+func NewAwsClientBuilder(client client.Client) *awsClientBuilder {
+	return &awsClientBuilder{
+		client: client,
+	}
+}
+
+func (b *awsClientBuilder) BuildSnowAwsClientMap(ctx context.Context) (aws.Clients, error) {
+	credentials, certificates, err := SnowCredentials(ctx, b.client)
+	if err != nil {
+		return nil, fmt.Errorf("getting snow credentials: %v", err)
+	}
+
+	return createAwsClients(ctx, credentials, certificates)
+}
+
+type credentialConfiguration struct {
+	AccessKey string `ini:"aws_access_key_id"`
+	SecretKey string `ini:"aws_secret_access_key"`
+	Region    string `ini:"region"`
+}
+
+func createAwsClients(ctx context.Context, credentials []byte, certificates []byte) (aws.Clients, error) {
```

Comment:
This seems to be similar to `BuildSnowAwsClientMap`. Is there a way to reuse code do they have to work very differently?

Reply:
As described in the above comment, I couldn't find a way to reuse the logic so I decided to keep it completely separate until we have a better idea on how to share the logic.